apps/03TicTacToe.js

```diff
@@ -25,22 +25,65 @@ function printBoard() {
 
 function horizontalWin() {
   // Your code here
+  if (board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn){
+  return true;
+  }
+  if (board[1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn){
+  return true;
+  }
+  if (board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn){
+  return true;
+  }
+  return false;
 }
 
 function verticalWin() {
   // Your code here
+  if (board[0][0] === playerTurn && board[1][0] === playerTurn && board[2][0] === playerTurn){
+  return true;
+  }
+  if (board[0][1] === playerTurn && board[1][1] === playerTurn && board[2][1] === playerTurn){
+  return true;
+  }
+  if (board[0][2] === playerTurn && board[1][2] === playerTurn && board[2][2] === playerTurn){
+  return true;
+  }
+  return false;
 }
 
 function diagonalWin() {
   // Your code here
+  if (board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] === playerTurn){
+  return  true;
+  }
+  if (board[2][0] === playerTurn && board[1][1] === playerTurn && board[0][2] === playerTurn){
+  return true;
+  }
+  return false;
 }
 
 function checkForWin() {
   // Your code here
+  if(horizontalWin() || verticalWin() || diagonalWin()){
+    console.log("Player" + playerTurn + "wins!");
+    return true;
+  }
 }
 
 function ticTacToe(row, column) {
   // Your code here
+
+  playerTurn =( board[row][column] = playerTurn );
+
+  function togglePlayer() {
+  playerTurn = (playerTurn === 'X') ? 'O' : 'X';
+  }
```

Comment:
`function togglePlayer(){...}` should be hosted outside your main (TicTacToe) function unless you set it up as a method.